rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -382,7 +393,7 @@ else if (isCached.get(idx).getLocation() == Location.CACHED) {
         catch (IOException e) {
           log.error("Unable to open file channel in R mode", e);
           // reset bookkeeper client so that we take direct route
-          this.bookKeeperClient = null;
+          bookKeeperClient = null;
```

Comment:
This will not do the intended work, setting argument to null will not set the value null at caller.

Reply:
Doing this was not the right thing to do at the first place, we are not closing the underlying transport which can cause leaks. 
We set `bookKeeperClient = null` to force next rounds of reads to fallback to direct reads when we see FS permission issues in opening files. I suggest we change `bookKeeperClient = null;` logic to instead use a variable in the CachingInputStream to mark permission errors, against which we would fallback to direct reads.

Reply:
Will raise an issue for this